src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -37,82 +37,94 @@ import QueryAndDownloadTabs from "shared/components/query/QueryAndDownloadTabs";
-                   return;
-                }
-                runInAction(()=>{
-                    // set query and pathname separately according to which changed, to avoid unnecessary
-                    //  recomputation by updating the query if only the pathname changed
-                    if (queryChanged) {
-                        // update query
-                        // normalize cancer_study_list this handles legacy sessions/urls where queries with single study had different param name
-                        const cancer_study_list = query.cancer_study_list || query.cancer_study_id;
-
-                        const cancerStudyIds: string[] = cancer_study_list.split(",");
-
-                        const oql = decodeURIComponent(query.gene_list);
-
-                        let samplesSpecification = parseSamplesSpecifications(query, cancerStudyIds);
-
-                        const changes = updateResultsViewQuery(resultsViewPageStore.rvQuery, query, samplesSpecification, cancerStudyIds, oql);
-                        if (changes.cohortIdsList) {
-                            resultsViewPageStore.initDriverAnnotationSettings();
-                        }
-
-                        onMobxPromise(resultsViewPageStore.studyIds, ()=>{
-                            try {
-                                trackQuery(resultsViewPageStore.studyIds.result!, oql, resultsViewPageStore.hugoGeneSymbols, resultsViewPageStore.queriedVirtualStudies.result!.length > 0);
-                            } catch {};
-                        });
-
-                        lastQuery = query;
-                    }
-                    if (pathnameChanged) {
-                        // need to set tab like this instead of with injected via params.tab because we need to set the tab
-                        //  at the same time as we set the query parameters, otherwise we get race conditions where the tab
-                        //  we're on at the time we update the query doesnt get unmounted because we change the query, causing
-                        //  MSKTabs unmounting, THEN change the tab.
-                        const tabId = getTabId(pathname);
-                        if (resultsViewPageStore.tabId !== tabId) {
-                            resultsViewPageStore.tabId = tabId;
-                        }
-                        lastPathname = pathname;
-                    }
-                });
-            }
-        },
-        {fireImmediately: true}
-    );
-
-    resultsViewPageStore.queryReactionDisposer = queryReactionDisposer;
+import ResultsViewURLWrapper from "pages/resultsView/ResultsViewURLWrapper";
+import setWindowVariable from "shared/lib/setWindowVariable";
+
+function initStore(appStore:AppStore, urlWrapper:ResultsViewURLWrapper) {
+
+    const resultsViewPageStore = new ResultsViewPageStore(appStore, getBrowserWindow().globalStores.routing, urlWrapper);
+
+    // whenever study list changes, reinit driver annotation settings
```

Comment:
@adamabeshouse  i leave this for reference in this review because it's important to make sure we are accomplishing all this in new model